unittests/allplatformstests.py

```diff
@@ -3868,3 +3869,166 @@ def do_install(tags=None):
         self.assertEqual(sorted(expected_custom), do_install('custom'))
         self.assertEqual(sorted(expected_runtime_custom), do_install('runtime,custom'))
         self.assertEqual(sorted(expected_all), do_install())
+
+    def test_introspect_install_plan(self):
+        testdir = os.path.join(self.unit_test_dir, '99 install_plan')
+        introfile = os.path.join(self.builddir, 'meson-info', 'intro-install_plan.json')
+        self.init(testdir)
+        self.assertPathExists(introfile)
+        with open(introfile, encoding='utf-8') as fp:
+            res = json.load(fp)
+
+        env = get_fake_env(testdir, self.builddir, self.prefix)
+
+        def output_name(name, type_):
+            return type_(name=name, subdir=None, subproject=None,
+                         for_machine=MachineChoice.HOST, sources=[],
+                         objects=[], environment=env, kwargs={}).filename
+
+        shared_lib_name = lambda name: output_name(name, SharedLibrary)
+        static_lib_name = lambda name: output_name(name, StaticLibrary)
+        exe_name = lambda name: output_name(name, Executable)
```

Comment:
wow, this is much smarter than the crap I did in test_install_tag(). Would be nice to move those helpers into the base class BasePlatformTests.

Reply:
Sure, but I would prefer to do that in a separate PR. Also, I think there are some quirks with this approach that we do not hit in this particular test, so I wouldn't say this would necessarily replace all similar usages, like the one in `test_install_tag`, but we can certainly replace a lot of it.

Reply:
(I don't mean this wouldn't replace the usage in `test_install_tag`, just that there may be some similar usages that it may not able to replace)

Reply:
Can be done in another PR indeed, but to keep things maintainable we need to make sure we factor common code into helper functions, so please don't forget to open one :)